homeassistant/components/ambient_network/config_flow.py

```diff
@@ -0,0 +1,169 @@
+        .get(API_STATION_LOCATION)
+    )
+    station_type = station.get(API_LAST_DATA, {}).get(API_STATION_TYPE)
+    return f"{location}{'' if location is None or station_type is None else ' '}{station_type}"
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle the config flow for the Ambient Weather Network integration."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Construct the config flow."""
+
+        self._longitude = 0.0
+        self._latitude = 0.0
+        self._radius = 0.0
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step to select the location."""
+
+        if user_input:
+            self._latitude = user_input[CONF_LOCATION][CONF_LATITUDE]
+            self._longitude = user_input[CONF_LOCATION][CONF_LONGITUDE]
+            self._radius = DistanceConverter.convert(
+                user_input[CONF_LOCATION][CONF_RADIUS],
+                UnitOfLength.METERS,
+                UnitOfLength.MILES,
+            )
+            return await self.async_step_station()
+
+        schema: vol.Schema = self.add_suggested_values_to_schema(
+            vol.Schema(
+                {
+                    vol.Required(
+                        CONF_LOCATION,
+                    ): LocationSelector(LocationSelectorConfig(radius=True)),
+                }
+            ),
+            {
+                CONF_LOCATION: {
+                    CONF_LATITUDE: self.hass.config.latitude,
+                    CONF_LONGITUDE: self.hass.config.longitude,
+                    CONF_RADIUS: CONF_RADIUS_DEFAULT,
+                }
+            },
+        )
+
+        return self.async_show_form(
+            step_id=CONF_USER,
+            data_schema=schema,
+        )
+
+    async def async_step_station(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the second step to select the station."""
+
+        if user_input:
+            mac_address, station_name = user_input[CONF_STATION].split(",")
+            await self.async_set_unique_id(mac_address)
+            self._abort_if_unique_id_configured()
+            return self.async_create_entry(
+                title=station_name,
+                data={
+                    CONF_MAC: mac_address,
+                },
+            )
+
+        client: OpenAPI = OpenAPI()
+        stations: list[dict[str, Any]] = await client.get_devices_by_location(
+            self._latitude, self._longitude, radius=self._radius
+        )
+
+        # Filter out indoor stations
+        stations = list(
+            filter(
+                lambda station: not station.get(API_STATION_INFO, {}).get(
+                    API_STATION_INDOOR, False
+                ),
+                stations,
+            )
+        )
+
+        if not stations:
+            return self.async_abort(reason="no_stations_found")
```

Comment:
Is this a good user experience to just abort if we don't find any stations?

Reply:
Fixed. We are now staying on the same form and show an error instead.